packages/core-data/src/actions.js

```diff
@@ -543,6 +558,34 @@ export function* saveEntityRecord(
 	}
 }
 
+export function* __experimentalBatchSaveEntityRecords( spec ) {
+	const batch = __experimentalCreateBatch( batchProcessor );
+
+	for ( const { kind, name, recordId, record } of spec ) {
+		if ( recordId ) {
+			dispatch( 'core' ).saveEditedEntityRecord( kind, name, recordId, {
+				__experimentalBatch: batch,
+			} );
+		} else if ( record ) {
+			dispatch( 'core' ).saveEntityRecord( kind, name, record, {
+				__experimentalBatch: batch,
+			} );
+		} else {
+			throw new Error(
+				"saveEntityRecords: 'recordId' or 'record' is required."
+			);
+		}
+	}
+
+	yield __unstableAwaitPromise( batch.waitForSize( spec.length ) );
```

Comment:
This is what fixes https://github.com/WordPress/gutenberg/issues/27173 properly.